### PR TITLE
fix: fix swapper feature flags

### PR DESCRIPTION
**Pull request by elmutt**
## Description

Fix feature flag toggle for swappers. 

One small bug that Im not sure why its happening but it also doesnt matter:

When toggling to enable a swapper and clicking "apply" it redirects to the dashboard instantly and the dropdowns for trading dropdowns dont load until you go to a different page and back to the dashboard. It could be something about redirecting to the dashboard too fast after the flag is enabled but i cant figure it out. It doesnt really matter because flags for only for internal testing and toggling works

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

### CODE REVIEW

File: src/components/Trade/hooks/useSwapper/useSwapper.ts [hunk 1]
```diff
@@ -111,7 +118,6 @@ const getSwapperManager = async (): Promise<SwapperManager> => {
         web3,
         adapter: avalancheChainAdapter,
       })
-
       _swapperManager.addSwapper(zrxAvalancheSwapper)
     }
     if (getConfig().REACT_APP_FEATURE_OSMOSIS) {
```
Comment (gomesalexandre) [inline src/components/Trade/hooks/useSwapper/useSwapper.ts:123]:
```suggestion
    if (flags.Osmosis) {
```

**Review by 0xdef1cafe — APPROVED** [review]

File: src/components/Trade/hooks/useSwapper/useSwapper.ts [hunk 2]
```diff
@@ -142,12 +148,12 @@ export const useSwapper = () => {
...
   // Swappers will be added in the useEffect below
   const [swapperManager, setSwapperManager] = useState<SwapperManager>(() => new SwapperManager())
-
+  const featureFlags = useSelector(selectFeatureFlags)
   useEffect(() => {
     ;(async () => {
       setSwapperManager(await getSwapperManager())
     })()
-  }, [])
+  }, [featureFlags])
```
Comment (gomesalexandre) [inline src/components/Trade/hooks/useSwapper/useSwapper.ts:156]: To quote our discussion on Discord:

> I can see why this works, as we get a new `useSwapper()` instance everytime we change trade routes, so you can't really the `useEffect()` hook in `useSwapper()` itself to react on a new reference

I don't think this does what it intends to i.e react on a new `featureFlags` reference to rerender the swapper.

You should be able to revert this hunk.

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Missing case for Osmosis, and one hunk that we can most likely revert, otherwise looks good

File: src/components/Trade/hooks/useSwapper/useSwapper.ts (hunk 2 above)
Comment (elmutt) [reply src/components/Trade/hooks/useSwapper/useSwapper.ts]: yep it seems to work fine without it. done

**Review by gomesalexandre — APPROVED** [review]
Tested again, working as expected for Thor, Avalanche and Osmosis swappers

🍫

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**